### PR TITLE
zwave introspection for commands

### PR DESCRIPTION
**Pull request by eljuguetero**

### CODE REVIEW

File: src/driver_zwave.cpp [hunk 1]
```diff
@@ -295,10 +296,43 @@ void ZWaveDriver::List() {
...
+        ZWaveMsg_ZWaveClassInfo* p_zwave_class = node->add_zwave_class();
+
+        p_zwave_class->set_zwave_class(zwave_class_type);
+
+        const zw_command_class* p_cmd_class =
+            zw_cmd_tool_get_class_by_name(class_name.c_str());
+
+        std::valarray<const char*> cmd_names(512);
+
+        int number_of_commands =
```
Comment (arhuaco) [inline src/driver_zwave.cpp:315]: const?

File: src/driver_zwave.cpp [hunk 2]
```diff
@@ -295,10 +296,43 @@ void ZWaveDriver::List() {
...
+            zw_cmd_tool_get_cmd_names(p_cmd_class, &cmd_names[0]);
+
+        for (auto& cmd_name : std::valarray<const char*>(
```
Comment (arhuaco) [inline src/driver_zwave.cpp:318]: const auto ?

File: src/driver_zwave.cpp [hunk 3]
```diff
@@ -295,10 +296,43 @@ void ZWaveDriver::List() {
...
+
+          std::valarray<const char*> param_names(512);
+
+          const zw_command* p_zw_command =
+              zw_cmd_tool_get_cmd_by_name(p_cmd_class, cmd_name);
+
+          int number_of_params =
+              zw_cmd_tool_get_param_names(p_zw_command, &param_names[0]);
+
+          for (auto& param_name : std::valarray<const char*>(
```
Comment (arhuaco) [inline src/driver_zwave.cpp:335]: const auto?

**Review by arhuaco — APPROVED** [review]

File: src/driver_zwave.cpp (hunk 2 above)
Comment (eljuguetero) [reply src/driver_zwave.cpp:318]: changed

File: src/driver_zwave.cpp (hunk 3 above)
Comment (eljuguetero) [reply src/driver_zwave.cpp:335]: changed

File: src/driver_zwave.cpp (hunk 1 above)
Comment (eljuguetero) [reply src/driver_zwave.cpp:315]: third party C function expect const char**  : 
 zw_cmd_tool_get_cmd_names(const struct zw_command_class* c, const char** names);

**Pull request merged**